### PR TITLE
Enable `-r/--region` command line args for `pcluster dcv connect` and `pcluster ssh`

### PR DESCRIPTION
**Pull request by tilne**
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: cli/pcluster/commands.py [hunk 1]
```diff
@@ -418,6 +418,8 @@ def ssh(args, extra_args):  # noqa: C901 FIXME!!!
     :param extra_args: pcluster CLI extra_args
     """
     # FIXME it always search for the default config file
+    if args.region:
+        os.environ["AWS_DEFAULT_REGION"] = args.region
```
Comment (tilne) [inline cli/pcluster/commands.py:422]: after looking at cli.py it seems this isn't needed.

File: cli/pcluster/createami.py [hunk 2]
```diff
@@ -308,8 +308,6 @@ def create_ami(args):
 
     # Ensure AWS_DEFAULT_REGION env is set.
     # Set it explicitly if the CLI arg was passed. Otherwise let the config object do it.
-    if args.region:
-        os.environ["AWS_DEFAULT_REGION"] = args.region
```
Comment (rexcsn) [inline cli/pcluster/createami.py:312]: I guess this code never worked, but createami command seems to accept a -r/--region option.
Should we remove that option from createami, or somehow make it work?

File: cli/pcluster/createami.py [hunk 3]
```diff
@@ -308,8 +308,6 @@ def create_ami(args):
 
     # Ensure AWS_DEFAULT_REGION env is set.
     # Set it explicitly if the CLI arg was passed. Otherwise let the config object do it.
```
Comment (hanwen-pcluste) [inline cli/pcluster/createami.py:310]: Shall we delete the comments?

**Review by hanwen-pcluste — APPROVED** [review]

File: cli/pcluster/createami.py (hunk 3 above)
Comment (tilne) [reply cli/pcluster/createami.py:310]: yes. good catch.

File: cli/pcluster/createami.py (hunk 2 above)
Comment (tilne) [reply cli/pcluster/createami.py:312]: i thought the same thing, bu this code was. redundant. there's a conditional in cli.py::main that does this for any subcommand that supports a region arg.

File: cli/pcluster/createami.py (hunk 3 above)
Comment (tilne) [reply cli/pcluster/createami.py:310]: done.

**Review by hanwen-pcluste — APPROVED** [review]

**Pull request merged**